src/sidebar/util/copy-to-clipboard.ts

```diff
@@ -30,3 +32,37 @@ export function copyText(text: string) {
     temp.remove();
   }
 }
+
+/**
+ * Copy the string `text` to the clipboard verbatim.
+ *
+ * @throws {Error}
+ *   This function may throw an error if the `clipboard-write` permission was
+ *   not allowed.
+ */
+export async function copyPlainText(text: string, navigator_ = navigator) {
+  await navigator_.clipboard.writeText(text);
+}
+
+/**
+ * Copy the string `text` to the clipboard, trying to preserve rich format.
+ *
+ * This is useful if provided text contains HTML, as copied content will include
+ * the result of rendering it, not the raw markup.
+ *
+ * If the browser does not support this, it will fall back to copy the string
+ * as plain text.
+ *
+ * @throws {Error}
+ *   This function may throw an error if the `clipboard-write` permission was
+ *   not allowed.
+ */
+export async function copyRichText(text: string, navigator_ = navigator) {
+  if (!navigator_.clipboard.write) {
+    await copyPlainText(text, navigator_);
```

Comment:
> I think we should try to avoid implementing this if we can though.

Agree. This is kind of a safety check, but according to the[ support list from MDN](https://developer.mozilla.org/en-US/docs/Web/API/Clipboard/write#browser_compatibility), only Firefox might have some problems dealing with this if it has not been manually enabled.

Other browsers have supported it out of the box for quite some time.

Reply:
>  only Firefox might have some problems dealing with this if it has not been manually enabled.

If the feature is not enabled out of the box, or via a preference exposed in non-developer UIs, then I would treat it as unsupported. Feature flags in browsers (eg. `about:config` in Firefox) is very much a developer-centric feature.